### PR TITLE
SL-2 [Creates auth strategy and adds apollo server]

### PR DESCRIPTION
**Pull request by krzysztofbogusz91**
SL-2 [Creates auth strategy and adds apollo server]

### CODE REVIEW

File: server/shema.js [hunk 1]
```diff
@@ -33,27 +60,74 @@ const RootQuery = new GraphQLObjectType({
...
-          type: GraphQLString
+        code: {type: GraphQLString},
+      },
+      resolve(parent, args) {
+        const code = args.code;
+        const secret = keys.sec;
+        const params = { client_id: '43111', client_secret: secret, code, grant_type: 'authorization_code'}
+        const url = new URL("https://www.strava.com/oauth/token" );
+        Object.keys(params).forEach(key => url.searchParams.append(key, params[key]));
+        return axios.post('' + url, {
```
Comment (slawomirkolodziej) [inline server/shema.js:85]: I think URL has `toString` method you can call.

File: client/src/components/ExchangePage.tsx [hunk 2]
```diff
@@ -1,37 +1,64 @@
...
+}
+const AUTH = gql`
+  query LoginData($code: String) {
+    auth (code: $code){
+      token,
+      refresh_token,
+      user {
+        firstname,
+        lastname,
+        profile,,
```
Comment (slawomirkolodziej) [inline client/src/components/ExchangePage.tsx:24]: Remove one `,`

File: client/src/components/ExchangePage.tsx [hunk 3]
```diff
@@ -1,37 +1,64 @@
...
-      const athlete = resps.user;
+  const { loading, error, data } = useQuery(AUTH, {
+    variables: { code }
+  });
+
+  const gotData = (authData: AuthData) => {
+      if(!authData) {
+        setTimeout(()=>{
+          props.history.push('/login')
+        }, 1)
```
[the remark this answers is not in the document]
Comment (krzysztofbogusz91) [reply client/src/components/ExchangePage.tsx:42]: For some reason  it breaks the app

File: server/server.js [hunk 4]
```diff
@@ -1,43 +1,36 @@
...
-app.get('/auth', function (req, res) {
-  const code = req.query.code
-  const secret = keys.sec;
-  const url = new URL("https://www.strava.com/oauth/token");
-  const params = { client_id: '43111', client_secret: secret, code, grant_type: 'authorization_code'}
- 
-  Object.keys(params).forEach(key => url.searchParams.append(key, params[key]));
+const axiosAuthMiddleware = (req, res, next) => {
+  if(!!req.headers['authorization']){
+    axios.defaults.headers.common['Authorization'] = req.headers['authorization'];
```
[the remark this answers is not in the document]
Comment (krzysztofbogusz91) [reply server/server.js:10]: Yes I have mised this fact, thanks its fixed.

**Pull request merged**